File: 4. CS50’s Introduction to Computer Science  - Harvard University/1. C/mario.c
#include <cs50.h>
#include <stdio.h>

//  Prompts user for a positive integer between 1 and 8, inclusive. Recreates Mario pyramid using hashes (#) for bricks.

int main(void)
{
    // Get height of pyramid from user as an interger
    int n;
    do
    {
        n = get_int("Height: ");
    }
    while (n <= 0 || n > 8);

    //printing the pyramid
    int m = n - 1;
    for (int i = 1; i <= n; i++)
    {
        // print required spaces
        for (int j = m; j > 0; j--)
        {
            printf(" ");
        }

        // Print #
        for (int k = 1; k <= i; k++)
        {
            printf("#");
        }

        printf("\n");
        m--;
    }

}

File: 4. CS50’s Introduction to Computer Science  - Harvard University/6. Python/sentimental-mario-less/mario.py
def main():
    # Get height from user
    height = get_int()

    # Set initial counter
    counter = 1

    # print the pyramid
    while counter <= height:
        print(" " * (height-counter), end="")
        print("#" * counter)
        counter += 1


def get_int():
    # Get interger from user else re-prompt until user input an interger between 1 and 8
    while True:
        try:
            height = int(input("Height: "))
            if 1 <= height <= 8:
                return height
        except ValueError:
            print("That is not an integer")
            pass


if __name__ == "__main__":
    main()

